File: URI-py/1040.py
# -*- coding: utf-8 -*-
n1 , n2, n3, n4 = map(float, input().split())
m = (n1*2+n2*3+n3*4+n4*1)/10

if m >= 7:
    print("Media: %.1f\nAluno aprovado" % m)
elif m <= 6.9 and m >= 5:
    print("Media: %.1f\nAluno em exame."%(m))
    ne = float(input())
    nf = (m + ne) / 2
    print("Nota do exame: %.1f"%ne)
    if nf == 5 or nf > 5:
        print("Aluno aprovado.\nMedia final: %.1f"%nf)

elif m < 5:
    print("Media: %.1f\nAluno reprovado" % m)
else:
    print("error")

File: URI-cpp/1040.cpp
#include <iostream>
#include <iomanip>
#include <math.h>

using namespace std;

int main()
{ 
    float a,b,c,d,average,grade,finalgrade;
    
    cin >> a >> b >> c >> d;
    cout << fixed << setprecision(1);

    average = (a*2+b*3+c*4+d*1)/10.0;
    if (average >= 7)
    {
        cout << "Media: " << average << endl;
        cout << "Aluno aprovado." << endl;
    }
    else if (average < 5)
    {
        cout << "Media: " << average << endl;
        cout << "Aluno reprovado." << endl;
    }
    else
    {
        cin >> finalgrade;
        
        cout << "Media: " << average << endl;
        cout << "Aluno em exame." << endl;
        cout << "Nota do exame: " << finalgrade << endl;

        if (((finalgrade+average)/2)>=5)
        {
            cout << "Aluno aprovado." << endl;
            cout << "Media final: " << (finalgrade+average)/2.0 << endl;
        }
        else
        {
            cout << "Aluno reprovado." << endl;
            cout << "Media final: " << (finalgrade+average)/2.0 << endl;
        }   
    }
    return 0;
}